src/main/java/org/alliancegenome/curation_api/services/validation/slotAnnotations/constructSlotAnnotations/ConstructFullNameSlotAnnotationValidator.java

```diff
@@ -0,0 +1,70 @@
+package org.alliancegenome.curation_api.services.validation.slotAnnotations.constructSlotAnnotations;
+
+import javax.enterprise.context.RequestScoped;
+import javax.inject.Inject;
+
+import org.alliancegenome.curation_api.dao.ConstructDAO;
+import org.alliancegenome.curation_api.dao.slotAnnotations.constructSlotAnnotations.ConstructFullNameSlotAnnotationDAO;
+import org.alliancegenome.curation_api.exceptions.ApiErrorException;
+import org.alliancegenome.curation_api.model.entities.Construct;
+import org.alliancegenome.curation_api.model.entities.VocabularyTerm;
+import org.alliancegenome.curation_api.model.entities.slotAnnotations.constructSlotAnnotations.ConstructFullNameSlotAnnotation;
+import org.alliancegenome.curation_api.response.ObjectResponse;
+import org.alliancegenome.curation_api.services.validation.slotAnnotations.NameSlotAnnotationValidator;
+
+@RequestScoped
+public class ConstructFullNameSlotAnnotationValidator extends NameSlotAnnotationValidator<ConstructFullNameSlotAnnotation> {
+
+	@Inject
+	ConstructFullNameSlotAnnotationDAO constructFullNameDAO;
+	@Inject
+	ConstructDAO constructDAO;
+
+	public ObjectResponse<ConstructFullNameSlotAnnotation> validateConstructFullNameSlotAnnotation(ConstructFullNameSlotAnnotation uiEntity) {
+		ConstructFullNameSlotAnnotation fullName = validateConstructFullNameSlotAnnotation(uiEntity, false, false);
+		response.setEntity(fullName);
+		return response;
+	}
+
+	public ConstructFullNameSlotAnnotation validateConstructFullNameSlotAnnotation(ConstructFullNameSlotAnnotation uiEntity, Boolean throwError, Boolean validateConstruct) {
+
+		response = new ObjectResponse<>(uiEntity);
+		String errorTitle = "Could not create/update ConstructFullNameSlotAnnotation: [" + uiEntity.getId() + "]";
+
+		Long id = uiEntity.getId();
+		ConstructFullNameSlotAnnotation dbEntity = null;
+		Boolean newEntity;
+		if (id != null) {
+			dbEntity = constructFullNameDAO.find(id);
+			newEntity = false;
+			if (dbEntity == null) {
+				addMessageResponse("Could not find ConstructFullNameSlotAnnotation with ID: [" + id + "]");
+				throw new ApiErrorException(response);
+			}
+		} else {
+			dbEntity = new ConstructFullNameSlotAnnotation();
+			newEntity = true;
+		}
+		dbEntity = (ConstructFullNameSlotAnnotation) validateNameSlotAnnotationFields(uiEntity, dbEntity, newEntity);
+
+		VocabularyTerm nameType = validateFullNameType(uiEntity.getNameType(), dbEntity.getNameType());
+		dbEntity.setNameType(nameType);
+
+		if (validateConstruct) {
+			Construct singleConstruct = validateSingleConstruct(uiEntity.getSingleConstruct(), dbEntity.getSingleConstruct());
+			dbEntity.setSingleConstruct(singleConstruct);
+		}
+
+		if (response.hasErrors()) {
+			if (throwError) {
+				response.setErrorMessage(errorTitle);
+				throw new ApiErrorException(response);
+			} else {
+				return null;
+			}
+		}
+
+		return dbEntity;
```

Comment:
It's maybe not as copy-and-paste as it looks.  Each name type validates against a different vocabularyTermSet (or Vocabulary rather than set in the case of Synonyms).  All the actual validation code is already elsewhere in shared methods.